CONTRIBUTING.md

```diff
@@ -0,0 +1,338 @@
+   Does the code change require an update of the documentation?
+
+#### Design
+
+- Does this change belong in the codebase?
+- Is it integrated well?
+
+#### Functionality
+
+- Does the code do what the developer intended?
+- Are there edge cases, where it could break?
+- For UI changes: give it a try yourself! (difficult to grasp from reading code)
+
+#### Complexity
+
+- Any complex lines, functions, classes that are not easy to understand?
+- Over-engineering: is the code too complex for the problem at hand?
+
+#### Tests
+
+- Are there tests for new functionality? \
+Are bugs covered by a test that breaks if the bug resurfaces?
+- Are the tests correct and useful? \
+Do they make simple and useful assertions?
+
+#### Naming
+
+- A good name is long enough to communicate what the item does, without being so long that it becomes hard to read
+
+#### Comments
+
+- Do comments explain *why* code exists (rather than *what* it is doing)?
+
+#### Style & Consistency
+
+- Does the contribution follow generic coding style (mostly enforced automatically)?
+
+(dev/merge_pr)=
+
+## Merging Pull Requests
+
+A pull request
+
+- should be opened only once you consider it ready for review.
+   Each time you push a commit to a branch with an open PR, it triggers a CI build, eating up the quota of the organization.
+- can consist of one or multiple commits.
+   Before you open a PR, make sure to clean up your commit history and create the commits that you think best divide up the total work as outlined above (use `git rebase` and `git commit --amend`).
+
+There are three ways of 'merging' pull requests on GitHub:
+
+- **Squash and merge**: take all commits, squash them into a single one and put it on top of the base branch.
+  - Choose this by default for pull requests that address a single issue and are well represented by a single commit.
+  - Make sure to clean the commit message (title & body)
+- **Rebase and merge**: take all commits and 'recreate' them on top of the base branch. All commits will be recreated with new hashes.
+  - Choose this for pull requests that require more than a single commit.
+  - Examples: PRs that contain multiple commits with individually significant changes; PRs that have commits from different authors (squashing commits would remove attribution)
+- **Merge with merge commit**: put all commits as they are on the base branch, with a merge commit on top
+  - Choose for collaborative PRs with many commits.
+     Here, the merge commit provides actual benefits.
+
+For the latter two, it is recommend that the merger do (if it is their PR) or at ask the contributor to perform an [interactive rebase]( https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history). At this point you can squash consecutive commits together (using "fixup") and also rewrite commit messages (using "reword") to create a compliant history.
```

Comment:
I feel like for these we should just tell people that we `squash and merge` for all repositories rather than have this extra information

Reply:
Err, I think there are legitimate use cases for the other two, but perhaps we could emphasize more that `squash and merge` is the most default/general case

Reply:
Added in dbfb8d56348c00b62b00b6145b0acdc72790ca81